hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/TokenRejectSuite.java

```diff
@@ -0,0 +1,556 @@
+                                .payingWith(TOKEN_TREASURY)
+                                .signedBy(TOKEN_TREASURY)
+                                .hasKnownStatus(ACCOUNT_IS_TREASURY),
+                        tokenReject(TOKEN_TREASURY, rejectingNFT(NON_FUNGIBLE_TOKEN_B, 1L))
+                                .hasKnownStatus(ACCOUNT_IS_TREASURY),
+                        tokenReject(TOKEN_TREASURY, rejectingToken(FUNGIBLE_TOKEN_A))
+                                .hasKnownStatus(ACCOUNT_IS_TREASURY),
+                        // remove Treasury's balance for Fungible_TOKEN-A and try again
+                        cryptoTransfer(moving(750L, FUNGIBLE_TOKEN_A).between(TOKEN_TREASURY, ACCOUNT)),
+                        tokenReject(TOKEN_TREASURY, rejectingToken(FUNGIBLE_TOKEN_A))
+                                .hasKnownStatus(ACCOUNT_IS_TREASURY),
+                        // Payer account does not own some of the NFTs he is rejecting
+                        tokenReject(
+                                        rejectingToken(FUNGIBLE_TOKEN_A),
+                                        rejectingToken(FUNGIBLE_TOKEN_B),
+                                        rejectingNFT(NON_FUNGIBLE_TOKEN_A, 2L),
+                                        rejectingNFT(NON_FUNGIBLE_TOKEN_B, 1L))
+                                .signedBy(ACCOUNT)
+                                .payingWith(ACCOUNT)
+                                .hasKnownStatus(INSUFFICIENT_TOKEN_BALANCE),
+                        // Account does not own the NFT he is rejecting
+                        tokenReject(ACCOUNT, rejectingToken(FUNGIBLE_TOKEN_A), rejectingNFT(NON_FUNGIBLE_TOKEN_A, 2L))
+                                .hasKnownStatus(INVALID_OWNER_ID),
+                        // Deleted tokens cannot be rejected, as you can already dissociate from them.
+                        tokenDelete(FUNGIBLE_TOKEN_A),
+                        tokenReject(ACCOUNT, rejectingToken(FUNGIBLE_TOKEN_A), rejectingNFT(NON_FUNGIBLE_TOKEN_A, 1L))
+                                .hasKnownStatus(TOKEN_WAS_DELETED))))
+                .then(
+                        getTokenNftInfo(NON_FUNGIBLE_TOKEN_A, 2L)
+                                .hasAccountID(TOKEN_TREASURY)
+                                .hasNoSpender(),
+                        getTokenNftInfo(NON_FUNGIBLE_TOKEN_A, 1L)
+                                .hasAccountID(ACCOUNT)
+                                .hasNoSpender(),
+                        getTokenNftInfo(NON_FUNGIBLE_TOKEN_B, 1L)
+                                .hasAccountID(TOKEN_TREASURY)
+                                .hasNoSpender(),
+                        getAccountBalance(TOKEN_TREASURY).logged().hasTokenBalance(FUNGIBLE_TOKEN_A, 0L),
+                        getAccountBalance(TOKEN_TREASURY).logged().hasTokenBalance(FUNGIBLE_TOKEN_B, 750L));
+    }
+
+    @HapiTest
+    final Stream<DynamicTest> tokenRejectFailsWithInvalidBodyInputsScenarios() {
```

Comment:
We should probably add a test that validates fees

Reply:
Done. I've created test that validates the fees and compares them to cryptoTransfer fees.